### ISSUE TITLE
Incompatibility with Datashader.by 

### ISSUE THREAD

**Issue by hoorayphyer** [holoviz/holoviews#4420]
The exciting development of “by” reduction operator of Datashader is not supported as a valid argument to “aggregator” of the HoloViews.datashade. The main issue seems to be about the member “column” enforced in HoloViews.datashade, whereas Datashader.by only has member "columns". It is desirable to incorporate "by" into the workflow of HoloViews.

#### ALL software version info
HoloViews version : 1.13.2
Datashader version : 0.11.0a4

#### Description of expected behavior and the observed behavior
Expected to produce plots when called with Datashader.by. Observed to give errors on member "column" not found on Datashader.by objects.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import holoviews as hv
hv.notebook_extension('bokeh')
import holoviews.operation.datashader as hd
import datashader as ds
import pandas as pd

df = pd.DataFrame({"x" : [0,1,0,1], "y" : [0,0,1,1], "cat" : [0,1,0,1], "weight" : [0,1,2,3]})
df["cat"] = df["cat"].astype("category")
points = hv.Points(df, kdims=["x","y"], vdims=["cat","weight"])
hd.datashade(points, aggregator=ds.by('cat',ds.sum("weight")) )
```

#### Stack traceback and/or browser JavaScript console output
![datashader](https://user-images.githubusercontent.com/8516409/81820619-d6870b80-94fe-11ea-92df-fcf06d654983.png)

**Comment by jbednar** [comment]
I've got various fixes for `by` on the `spanfixes` branch in a [Datashader PR](https://github.com/holoviz/datashader/pull/910). and I've just pushed [one more](https://github.com/holoviz/datashader/pull/910/commits/4a66da0c5565cf00ae6a541a968e4c431cbfa622) that makes self.column be a property returning the category column.  Given that `by` is unreleased, presumably you are using Datashader from git, and if so it would be great if you could test whether this change is sufficient.

**Comment by hoorayphyer** [comment]
@jbednar Actually I was using Datashader installed through `conda install -c pyviz/label/dev datashader`. But I tried the git version. The way I did follows the developer instructions given [here](https://datashader.org/getting_started/index.html#installation), except that before going from 2 to 3, I did `git checkout spanfixes`. The error message is now different. Please check below if the printed `ds_version` was the correct one, in case I did something wrong.
![Screenshot_2020-05-13 Untitled](https://user-images.githubusercontent.com/8516409/81852600-4ad6a480-9529-11ea-9fff-104942cb4a6a.png)

**Comment by jbednar** [comment]
Ah, ok. I made a slight update to the `spanfixes` branch and pushed it, but either way I'm able to reproduce your problem, and moreover I can reproduce it just with a `count()` aggregate that should work, given that `count_cat()` works:

```
import holoviews as hv, holoviews.operation.datashader as hd
import datashader as ds, pandas as pd
hv.extension('bokeh')

df = pd.DataFrame({"x" : [0,1,0,1], "y" : [0,0,1,1], "cat" : [0,1,0,-1], "weight" : [0,1,2,32]})
df["cat"] = df["cat"].astype("category")
points = hv.Points(df, kdims=["x","y"], vdims=["cat","weight"])
hd.datashade(points, aggregator=ds.count_cat("cat"), x_sampling=0.2, y_sampling=0.2)
```

![image](https://user-images.githubusercontent.com/1695496/81870134-e1f82800-953a-11ea-8f10-8dc879156e41.png)

```
hd.datashade(points, aggregator=ds.by("cat", ds.count()), x_sampling=0.2, y_sampling=0.2)
```
![image](https://user-images.githubusercontent.com/1695496/81870206-02c07d80-953b-11ea-9b4f-fc192fe6b4ea.png)

```
hd.datashade(points, aggregator=ds.by("cat", ds.sum("weight")), x_sampling=0.2, y_sampling=0.2)
```
![image](https://user-images.githubusercontent.com/1695496/81870248-1966d480-953b-11ea-8f18-86dac867ecac.png)

**Comment by jbednar** [comment]
Looks like holoviews will need some updates, but it would be nice to have those worked out before we release Datashader, in case Datashader also needs updates.

**Comment by philippjfr** [comment]
Fixed in #4438

**Comment by hoorayphyer** [comment]
> Fixed in #4438

Thanks, Philipp! It came just in time for me. So do I just `conda upgrade holoviews` to enjoy the fix or I need to build it from git?

**Comment by philippjfr** [comment]
I'll have a build up on the pyviz/label/dev conda channel later today but you can install straight from git as well if you need it asap.
